spdx/v3/v3_0/interfaces_only/model.go

```diff
@@ -0,0 +1,342 @@
+package v3_0
+
+import "time"
+
+type ElementID string
+
+type SpdxDocument interface {
+	ElementCollection
+}
+
+type spdxDocument struct {
+	// Element properties
+	spdxID ElementID
+	name   string
```

Comment:
Generating each abstract parents' properties using jinja would be ok to do. You think this is a better option than using a struct called `elementProperties` (purposefully not exporting the struct, but making the concrete class implement the interface `Element`).